jobs/resource_job.go

```diff
@@ -574,6 +574,21 @@ func (a JobsAPI) Read(id string) (job Job, err error) {
 		job.Settings.sortTasksByKey()
 		job.Settings.sortWebhooksByID()
 	}
+
+	if job.RunAsUserName != "" && job.Settings != nil {
+		userNameIsEmail := strings.Contains(job.RunAsUserName, "@")
+
+		if userNameIsEmail {
+			job.Settings.RunAs = &JobRunAs{
+				UserName: job.RunAsUserName,
+			}
+		} else {
+			job.Settings.RunAs = &JobRunAs{
+				ServicePrincipalName: job.RunAsUserName,
```

Comment:
@dvkuznietsov  Two questions: 
* Is it certain that if a job is not userEmail then it is service principal? 
* Is checking for @ for email enough? 

Hi @nkvuong, do you know if there is some existing utility for checking valid email in terraform?

Reply:
> Is it certain that if a job is not userEmail then it is service principal?
> Is checking for @ for email enough?

as of now, run_as_user_name is either an email or a UUID. String parsing is not ideal and that's why we moved on to run_as field in write path, but that's what we work with for now

Reply:
and, to answer your question, I beleive that's enough

Reply:
@dvkuznietsov is it too much to ask the Jobs API to use run_as for reads and writes rather than force clients to do this mapping themselves? Thinking RESTfully, when a user sets a field on a resource, they would expect to get that field as they set it when getting that resource back. However, here, Jobs may be breaking that pattern. WDYT?

Reply:
@mgyucht the fields are not symmetric: run_as is an optional field that we have recently added and run_as_user_name is an old field that is almost always filled in.

The intention for this change is if users don't provide run_as, we don't need to catch drifts in it, but if they do provide it, we do need to detect drifts

Reply:
Can you include `run_as` in the get response though? I guess what I don't understand is why it is a write-only field.

Reply:
we can't. It would be a breaking change for customers that pipe GET responses to create/reset calls

Reply:
Forgive me for asking, but why would that be a breaking change? Does the Jobs API not tolerate both of these parameters being set at the same time in a create/reset call?